logback-core/src/test/java/ch/qos/logback/core/net/server/WaitingClient.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Logback: the reliable, generic, fast and flexible logging framework.
+ * Copyright (C) 1999-2013, QOS.ch. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation.
+ */
+package ch.qos.logback.core.net.server;
+
+import ch.qos.logback.core.net.server.Client;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+
+/**
+ * A mock {@link Client} that notifies waiting thread when it has started,
+ * and waits to be interrupted before exiting.
+ *
+ * @author Carl Harris
+ */
+class WaitingClient implements Client {
+
+
+  private final Lock lock = new ReentrantLock();
+  private final Condition startCondition = lock.newCondition();
+  private final Condition stopCondition = lock.newCondition();
+
+  private boolean running;
+  private boolean closed;
+
+  public void run() {
+    running = true;
+    signalStartCondition();
+    while (running && !Thread.currentThread().isInterrupted()) {
+      try {
+        awaitStopCondition(Integer.MAX_VALUE);
+      } catch (InterruptedException ex) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  public void close() {
+    running = false;
+    closed = true;
+    signalStopCondition();
+  }
+
+  void awaitStartCondition(long timeout) throws InterruptedException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/tony19/logback/file/2376051824/issues/source?bid=3222618&fileBranchId=3222711#l58)